File: cifrado/escritorio/codigo/Ruby/Cifrado/CifradoCesar.rb
class CifradoCesar

	def initialize saltos, alfabeto
		@saltos = saltos
		@alfabeto = alfabeto
	end

	def modificarSaltos nuevoSalto
		@saltos = nuevoSalto
	end

	def cifrar textoPlano
		respuesta=""
		textoPlano.textoPlano.length.times do |i|
		   	respuesta += (@alfabeto.include?textoPlano.textoPlano[i]) ? @alfabeto[(posicion(textoPlano.textoPlano[i])+@saltos)% @alfabeto.length] : textoPlano.textoPlano[i]
		end
		respuesta
	end

	private

	def posicion c
		@alfabeto.index(c)
	end

end

File: cifrado/escritorio/codigo/PHP/Cifrado/CifradoCesar.php
<?php 
class CifradoCesar{

	public $saltos,$alfabeto;

	function __construct($saltos,$alfabeto){
		$this->saltos = $saltos;
		$this->alfabeto = $alfabeto;
	}

	function modificarSaltos($nuevoSalto){
		$this->saltos = $nuevoSalto;
	}


	function cifrar($textoPlano){
		$respuesta = "";
    	for ($i=0; $i < strlen($textoPlano->textoPlano) ; $i++) { 
    		$index = ($this->posicion($textoPlano->textoPlano[$i])+$this->saltos)%strlen($this->alfabeto);
    		$respuesta .= strpos($this->alfabeto,$textoPlano->textoPlano[$i])!==false? $this->alfabeto[$index] : $textoPlano->textoPlano[$i];
    	}
    	return $respuesta;
	}

	private function posicion($c){
    	return strpos($this->alfabeto,$c);
	}
	
}
?>